src/app/Library/CrudPanel/Traits/ColumnsProtectedMethods.php

```diff
@@ -132,6 +128,46 @@ protected function makeSureColumnHasWrapper($column)
         return $column;
     }
 
+    protected function makeSureColumnHasEntity($column)
+    {
+        if (isset($column['entity'])) {
+            return $column;
+        }
+
+        // if the name is an array it's definitely not a relationship
+        if (is_array($column['name'])) {
+            return $column;
+        }
+
+        //if the name is dot notation we are sure it's a relationship
```

Comment:
I just realised, maybe you need another check here? When I thought about assuming dot means relation, I figured out this isn't always true: if you have an array attribute on your model, you might want to use dot notation to add an array item as column.

Reply:
Hello @jnoordsij 

I agree with your proposed change. To be a relationship, your condition must be true.